### PR TITLE
Allow naming of different test class types

### PR DESCRIPTION
**Pull request by pokrakam**
Propose to use the more commonly used naming conventions for test classes rather than naming everything ltcl_. These are used by SAP and also referenced in the style guides e.g. the section on [Helper Methods](https://github.com/SAP/styleguides/blob/master/clean-abap/CleanABAP.md#put-help-methods-in-help-classes) and used in the ABAP unit testing book.

ltc_: test class
lth_: test helper
ltd_: test double
lts_: test suite

Oh, and also corrected bug in exception class name that I spotted when editing

### CODE REVIEW

**Comment by larshp** [comment]
guess its okay, I just dont know what defines each category, and should more be added to the list? injector, mock, stub, factory?

**Comment by mbtools** [comment]
Do we care? "^LT.*$"

File: abaplint.json [hunk 1]
```diff
@@ -215,9 +215,9 @@
       "ignoreExceptions": true
     },
     "local_class_naming": {
-      "exception": "^LCL_.*$",
+      "exception": "^LCX_.*$",
       "local": "^LCL_.*$",
-      "test": "^LTCL_.*$"
+      "test": "^LT(C|D|H|S|CL)_.*$"
```
Comment (larshp) [inline abaplint.json:220]: as suggested by Marc,

```suggestion
      "test": "^LT.+$"
```

**Comment by pokrakam** [comment]
The background for the choice of prefixes is less of my opinion but rather going with the SAP ASE recommendations in their book and the most common writing I've seen on the topic. 
The critical ones are: test classes which should be simple and descriptive, the mock/double classes which effectively represent test data, and everything else is helper. Subtler distinctions can be included in the name, `lth_widget_stub` as more complex naming may not be familiar to everyone.
But yeah, I guess as long as the prefix start with lt?_ it's OK.

**Pull request merged**